Add savepoints docs and other improvements to "saving" (#2698)
dotnet/EntityFramework.Docs: entity-framework/core/saving/transactions.md

## Default transaction behavior

By default, if the database provider supports transactions, all changes in a single call to `SaveChanges` are applied in a transaction. If any of the changes fail, then the transaction is rolled back and none of the changes are applied to the database. This means that `SaveChanges` is guaranteed to either completely succeed, or leave the database unmodified if an error occurs.

For most applications, this default behavior is sufficient. You should only manually control transactions if your application requirements deem it necessary.

## Controlling transactions

You can use the `DbContext.Database` API to begin, commit, and rollback transactions. The following example shows two `SaveChanges` operations and a LINQ query being executed in a single transaction:

[!code-csharp[Main](../../../samples/core/Saving/Transactions/ControllingTransaction.cs?name=Transaction&highlight=2,16-18)]

While all relational database providers support transactions, other providers types may throw or no-op when transaction APIs are called.

## Savepoints

> When `SaveChanges` is invoked and a transaction is already in progress on the context, EF automatically creates a *savepoint* before saving any data. Savepoints are points within a database transaction which may later be rolled back to, if an error occurs or for any other reason. If `SaveChanges` encounters any error, it automatically rolls the transaction back to the savepoint, leaving the transaction in the same state as if it had never started. This allows you to possibly correct issues and retry saving, in particular when [optimistic concurrency](xref:core/saving/concurrency) issues occur.

Reviewer: > This allows you to possibly correct 

What's *this* ?

See [Writing for translation](https://review.docs.microsoft.com/en-us/help/contribute/localization-mt-guidance?branch=master)

Short sentences should be typically about 15-20 words.

Author: `this` here is an indefinite anaphoric pronoun, referring in general to what has been mentioned in the previous sentence.

Re 15-20 words, I don't see any sentence in this paragraph which significantly exceeds that. IMO writing guidelines are definitely not meant as hard quantitative rules.

## Default transaction behavior

By default, if the database provider supports transactions, all changes in a single call to `SaveChanges` are applied in a transaction. If any of the changes fail, then the transaction is rolled back and none of the changes are applied to the database. This means that `SaveChanges` is guaranteed to either completely succeed, or leave the database unmodified if an error occurs.

For most applications, this default behavior is sufficient. You should only manually control transactions if your application requirements deem it necessary.

## Controlling transactions

You can use the `DbContext.Database` API to begin, commit, and rollback transactions. The following example shows two `SaveChanges` operations and a LINQ query being executed in a single transaction:

[!code-csharp[Main](../../../samples/core/Saving/Transactions/ControllingTransaction.cs?name=Transaction&highlight=2,16-18)]

While all relational database providers support transactions, other providers types may throw or no-op when transaction APIs are called.

## Savepoints

> When `SaveChanges` is invoked and a transaction is already in progress on the context, EF will automatically create a *savepoint* before saving any data. Savepoints are points within a database transaction which may be later be rolled back to, if an error occurs or for any other reason. If `SaveChanges` encounters any error, it will automatically roll the transaction back to the savepoint, leaving the transaction in the same state as if it had never started. This allows you to possibly correct issues and retry saving, in particular when [optimistic concurrency](xref:core/saving/concurrency) issues occur.

Reviewer: ```suggestion
When `SaveChanges` is invoked and a transaction is already in progress on the context, EF automatically creates a *savepoint* before saving any data. Savepoints are points within a database transaction which may be later be rolled back. For example, savepoints can roll back if an error occurs or for other reasons. If `SaveChanges` encounters an error:
* The transaction is automatically rolled back to the savepoint.
* The transaction stays in the same state as if it had never started.
Savepoints:
*  Allow the possibly to correct issues and retry saving.
* Are useful when [optimistic concurrency](xref:core/saving/concurrency) issues occur.
```

Reviewer: Avoid future tense and long sentences.

Author: Definitely accept the future tense comment, need to get used to that.

But I looked at the text again in preview, and the last two sentences don't seem overly long to me... IMHO it's clearer (and shorter) as-is than broken down into bullet points...
